packages/chat-core/src/task-state/TaskStateMachine.runtimeProbe.ts

```
@@ -69,6 +80,12 @@ export class RuntimeStabilityProbe {
       // Keep the unstable state. The state machine will re-arm the probe if it is still needed.
     } finally {
       this.running = false
+      const pendingReason = this.pendingReason
+      this.pendingReason = undefined
+      if (!this.isClosed() && pendingReason) {
+        await this.runNow(pendingReason)
+        return
```

Biome (2.5.0)

[error] Unsafe usage of 'return'.

(lint/correctness/noUnsafeFinally)